src/thor/worker.cc

```diff
@@ -140,17 +138,13 @@ namespace valhalla {
         }
 
         // Set the interrupt function
-        interrupt_callback = &interrupt;
+        service_worker_t::set_interrupt(interrupt_function);
 
         //flag healthcheck requests; do not send to logstash
         healthcheck = request.get<bool>("healthcheck", false);
         // Initialize request - get the PathALgorithm to use
         ACTION_TYPE action = static_cast<ACTION_TYPE>(request.get<int>("action"));
         boost::optional<int> date_time_type = request.get_optional<int>("date_time.type");
-        // Allow the request to be aborted
-        astar.set_interrupt(&interrupt);
-        bidir_astar.set_interrupt(&interrupt);
-        multi_modal_astar.set_interrupt(&interrupt);
```

Comment:
these get set downstream before the objects get used (commented above). setting them here in the work function means that they dont get set for the `actor` so we just take care of it down stream.